test/unit/org/apache/cassandra/utils/AbstractTypeGenerators.java

```diff
@@ -550,10 +550,11 @@ public static Gen<AbstractType<?>> safeTypeGen()
                                                         .withoutTypeKinds(COUNTER)
                                                         .withoutPrimitive(DecimalType.instance)
                                                         // its ordering is special...
-                                                        .withoutPrimitive(DurationType.instance);
-        // composite requires all elements fit into Short.MAX_VALUE bytes
-        // so try to limit the possible expansion of types
-        return baseline.withCompositeElementGen(new TypeGenBuilder(baseline).withDefaultSizeGen(1).withMaxDepth(1).build())
+                                                        .withoutPrimitive(DurationType.instance)
+                                                        // To make sure all elements fit within Short.MAX_VALUE bytes,
+                                                        // need to limit the possible expansion of types
+                                                        .withDefaultSizeGen(1).withMaxDepth(1);
```

Comment:
this makes sure collections/udt/tuple/vector/etc. are also bound.

The top level can be any type, but `list<list<int>>` is no longer possible; only `list<int>`

Reply:
this method is also only used for `partitioners` when figuring out the type for `LocalPartitioner`.  Given its isolation this change only impacts `LocalPartitioner` type and not tables